src/class_sqlThread.py

```diff
@@ -15,128 +15,122 @@
-            'bitmessagesettings', 'settingsversion')
-
-        # People running earlier versions of PyBitmessage do not have the
-        # usedpersonally field in their pubkeys table. Let's add it.
-        if settingsversion == 2:
-            item = '''ALTER TABLE pubkeys ADD usedpersonally text DEFAULT 'no' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 3
-
-        # People running earlier versions of PyBitmessage do not have the
-        # encodingtype field in their inbox and sent tables or the read field
-        # in the inbox table. Let's add them.
-        if settingsversion == 3:
-            item = '''ALTER TABLE inbox ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE inbox ADD read bool DEFAULT '1' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE sent ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 4
-
-        BMConfigParser().set(
-            'bitmessagesettings', 'settingsversion', str(settingsversion))
-        BMConfigParser().save()
+    @property
+    def sql_schema_version(self):
+        self._current_level = self.__get_current_settings_version()
+
+    @sql_schema_version.setter
+    def sql_schema_version(self, cur, conn, current_level):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        parameters = (current_level + 1,)
+        self.cur.execute(item, parameters)
+        self._current_level = self.__get_current_settings_version()
+
+    # @sql_schema_version.setter
+    def upgrade_to_latest(self, cur, conn, current_level=None):
+        """
+            Initialise upgrade level
+        """
+
+        # Declare variables
+        self.conn = conn
+        self.cur = cur
+        if current_level:
+            self._current_level = current_level
+        else:
+            self._current_level = self.__get_current_settings_version()
+        self.max_level = 11
+
+        # call upgrading level in loop
+        for l in range(self._current_level, self.max_level):
+            if int(l) == 1:
+                continue
+            self._upgrade_one_level_sql_statement(l)
+            # self.increment_settings_version(l)
+            self._current_level = l
+
+    def increment_settings_version(self, level):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        parameters = (level + 1,)
+        self.cur.execute(item, parameters)
 
-        helper_startup.updateConfig()
+    def embaded_version(self):
```

Comment:
typo

Reply:
should be called something like `init_database`

Reply:
Done

Reply:
Changes are done for that and, but for some query I applied same query  script.